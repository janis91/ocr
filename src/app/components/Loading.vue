<template>
  <div class="ocr-load">
    <span class="icon icon-loading"></span>
    <span>
      {{filesQueuedText}}
      <br />
      {{successfullyProcessedText}}
    </span>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { Translations } from '@a/configuration/Translations'

export default Vue.extend({
  name: 'Loading',
  props: {
    filesQueued: {
      type: Number,
      required: true
    },
    successfullyProcessed: {
      type: Number,
      required: true
    }
  },
  computed: {
    filesQueuedText(): string {
      return Translations.TRANSLATION_FILES_QUEUED(this.filesQueued)
    },
    successfullyProcessedText(): string {
      return Translations.TRANSLATION_FILES_SUCCESSFULLY_PROCESSED(String(this.successfullyProcessed), String(this.filesQueued))
    }
  }
})
</script>

<style scoped>
.ocr-load {
  display: flex;
  flex-direction: column;
  align-self: center;
  align-items: center;
  text-align: center;
}

.ocr-load > .icon.icon-loading {
  margin: 15px;
}
</style>
