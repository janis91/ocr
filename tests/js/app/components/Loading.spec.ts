import '../../GlobalMocks.mock'
import { mount, Wrapper } from '@vue/test-utils'
import Loading from '@a/components/Loading.vue'

describe('Loading.vue', () => {
  let wrapper: Wrapper<any>

  beforeEach(() => {
    wrapper = mount(Loading, {
      propsData: {
        filesQueued: 1,
        successfullyProcessed: 0
      }
    })
  })

  it('GIVEN standard props, WHEN Loading is mounted, THEN matches snapshot with translation for files queued and processed.', async () => {
    expect(wrapper.contains('span.icon-loading')).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })
})
