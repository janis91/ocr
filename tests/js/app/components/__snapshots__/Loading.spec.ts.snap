// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading.vue GIVEN standard props, WHEN Loading is mounted, THEN matches snapshot with translation for files queued and processed. 1`] = `
<div
  class="ocr-load"
>
  <span
    class="icon icon-loading"
  />
   
  <span>
    
    %n file is being processed:
    
    <br />
    
    {file}/{files} Files successfully processed
  
  </span>
</div>
`;
